### ISSUE TITLE
Make OLED content more intuitive

### ISSUE THREAD

**Issue by mondalaci** [UltimateHackingKeyboard/firmware#1005]
According to tester feedback, some parts of the OLED are not intuitive.

In place of the `+` / `-` / `%` charging state, let's show a [lightning icon](https://fontawesome.com/icons/bolt-lightning?f=classic&s=solid) when charging is in progress and `%` otherwise.

In place of the `N` and `C` num lock and caps lock indicators, let's try the following icons:

![Screenshot 2024-09-26 at 21 21 24](https://github.com/user-attachments/assets/faa20e1c-9908-44cc-a2d9-19b742a25354)

If `1` and `A` are too small to read well within the outer rectangles, using them without their rectangles may still be a better option than the current one.

(Originally reported by @rogermeletta.)

**Comment by mondalaci** [comment]
Please let's also try these plug icons instead of the current one:

![plug1](https://github.com/user-attachments/assets/2a3bcd76-01f7-437b-8bd1-f00f9a592762)
![plug2](https://github.com/user-attachments/assets/b92f4c80-4110-4fa2-8eb5-94cd91828289)

**Comment by kareltucek** [comment]
I would be confused by these as they indicate that the cable is disconnected.

**Comment by mondalaci** [comment]
How about this?

![connected](https://github.com/user-attachments/assets/5f96a0ae-7cc3-4848-870b-33df0bb57ac5)

**Comment by kareltucek** [comment]
Yeah, that could work.

**Comment by mhantsch** [comment]
I'd suggest you try to find an icon that looks more like a USB connector, or maybe just use the standard USB logo. The power plugs above look more like 230V AC plugs.

**Comment by mondalaci** [comment]
The keyboard halves are not connected via USB but with the spiral cable. Besides, this icon is very small on the display, and I doubt the suggested alternative would make a difference. I don't think there's a substantially better alternative but feel free to suggest a concrete icon.

**Comment by mhantsch** [comment]
Oh, I see now, you are indicating that the halves are connected, not that power comes via a connector! My misunderstanding.

I think the icon you proposed is good.

**Comment by rogermeletta** [comment]
Hm yeah it confused me too when I got the UHK80. 
What are you guys think about something like this to indicate the connection between the halves?
<img width="529" alt="Screenshot 2024-10-11 at 10 36 17@2x" src="https://github.com/user-attachments/assets/c80d8cd1-7016-4a45-ad9b-4b34882a0579">
(Just some mockup in GIMP)

**Comment by mondalaci** [comment]
I think the suggested icon is not very intuitive and reminds me of a barbell or jumping rope.

**Comment by rogermeletta** [comment]
or maybe like this?
<img width="408" alt="Screenshot 2024-10-11 at 12 17 55@2x" src="https://github.com/user-attachments/assets/fceb0c36-5a8c-43af-9780-829a6a75f420">

**Comment by mondalaci** [comment]
Thanks, this is much better.

@kareltucek Wanna give a shot?

**Comment by kareltucek** [comment]
I don't like it much either. 

I am happy to spend long hours on improving, testing various variants, drawing both pixel and svg graphics, and generally fiddling with it, but I would first like to get the pressing issues done.

Also, this is a relatively simple process. Anyone with basic C knowledge can fiddle with it (wink wink @mhantsch).

**Comment by mondalaci** [comment]
I agree that it's not a priority. Reopening this issue, so that we can address it later.

**Comment by kareltucek** [comment]
Also I have forgotten that battery-related icon...

**Comment by mhantsch** [comment]
I agree; keep the icons that you have right now; I think they are good enough and functional. 
At some point _in the future_, revisit and maybe work UI design in one consistent swoop.

**Comment by mhantsch** [comment]
Just because we were throwing in ideas for icons, I'm dropping this draft here.

![image](https://github.com/user-attachments/assets/982957c5-ca43-4cf5-9d13-afa0834ddcbe)

**Comment by kareltucek** [comment]
Btw., we are currently using FontAwesome kits, so I am not sure how much we actually want pixel graphics ideas. Apart from that, would have to see that implemented. 

I am also wondering if we should redesign the status bar and use the positioning to indicate that it is inter-half connection. Like just a simple "spiral cable" icon placed between the Left/Right battery indicators, or something like that.

**Comment by mondalaci** [comment]
> Btw., we are currently using FontAwesome kits, so I am not sure how much we actually want pixel graphics ideas. Apart from that, would have to see that implemented.

Font Awesome may not have the perfect icon for this purpose, so I consider using a custom icon justified.

> I am also wondering if we should redesign the status bar and use the positioning to indicate that it is inter-half connection. Like just a simple "spiral cable" icon placed between the Left/Right battery indicators, or something like that.

I love the idea, but the "battery included" product option will be optional, and if there are no batteries, no percentage values are shown on the display. I prefer a universal icon that can be used in all use cases.

**Comment by kareltucek** [comment]
> Font Awesome may not have the perfect icon for this purpose, so I consider using a custom icon justified.

I mean that current workflow is to create an svg icon, import it into our custom fontawesome kit, and just then export it into a lv-font-based font-file of ours.

I mean, it is just a convention that we don't need to stick to, but:

- Mixing pixel-aliased and pixel-unaliased icons could be very ugly. They are just designed differently.
- If we want to use the same icon sets for other purposes (e.g., show them in agent, on the web, or in some future product), then not having them all in one place and (vector-based) format may be inconvenient.

> I love the idea, but the "battery included" product option will be optional, and if there are no batteries, no percentage values are shown on the display. I prefer a universal icon that can be used in all use cases.

You know, that's a detail that can be solved in a number of ways.

**Comment by mondalaci** [comment]
> > Font Awesome may not have the perfect icon for this purpose, so I consider using a custom icon justified.
> 
> I mean that current workflow is to create an svg icon, import it into our custom fontawesome kit, and just then export it into a lv-font-based font-file of ours.
> 
> I mean, it is just a convention that we don't need to stick to, but:
> 
> * Mixing pixel-aliased and pixel-unaliased icons could be very ugly. They are just designed differently.
> * If we want to use the same icon sets for other purposes (e.g., show them in agent, on the web, or in some future product), then not having them all in one place and (vector-based) format may be inconvenient.

Yours are good points, and we should consider them. I don't insist on the suggested icons, but we can upscale them later.

> > I love the idea, but the "battery included" product option will be optional, and if there are no batteries, no percentage values are shown on the display. I prefer a universal icon that can be used in all use cases.
> 
> You know, that's a detail that can be solved in a number of ways.

I'm not arguing but wanted to provide a perspective on this issue.

**Comment by kareltucek** [comment]
@mondalaci I believe this is solved. Please correct me if not.
